extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-mysql/src/main/java/org/apache/guacamole/auth/mysql/conf/MySQLEnvironment.java

```diff
@@ -104,6 +104,12 @@ public class MySQLEnvironment extends JDBCEnvironment {
      * The default SSL mode for connecting to MySQL servers.
      */
     private final MySQLSSLMode DEFAULT_SSL_MODE = MySQLSSLMode.PREFERRED;
+    
+     /**
+     * The default maximum number of identifiers/parameters to be included in a 
+     * single batch when executing SQL statements.
+     */
+    private static final int DEFAULT_BATCH_SIZE = 2147483647;
```

Comment:
Is there documentation that might be referenced here that shows `2147483647` is the relevant limit for both MySQL and MariaDB?

Reply:
It looks like MySQL and MariaDB have a limitation on the maximum size of the SQL query, which is determined by the 1max_allowed_packet1 system variable. This variable defines the maximum packet size that the server can send or receive. I could change the code to using this value with some calculation to determine a batch size

Reply:
The value 2147483647 represents the maximum signed integer value in MySQL and MariaDB, which can be used as the maximum allowed number of parameters in a single SQL query. However, the actual limit might depend on the specific version and configuration of the database system. Instead of referring to the maximum signed integer value, it could be more appropriate to reference the practical limits of MySQL and MariaDB in this context.